web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -436,9 +463,109 @@ export class BarChartComponent implements AfterViewInit {
       .extent([[70, 0], [this.width - 90, this.heightBrush]]) // Limit the brush.
       .on('brush', brushDown); // Update the focus bar chart based on the brush selection.
 
-    this.brush.append('g')
+    this.brush
+      .append('g')
       .attr('class', 'brush')
       .call(brushSelector)
       .call(brushSelector.move, [firstItemPosition, lastItemPosition]);
   }
+
+  /**
+   * This function creates a boxplot next to the focus bar chart with all selected data.
+   * It indicates maximum, minimium, median, first quantile and third quantile of the
+   * selected data durations.
+   *
+   * @param inputData: Data selected for the focus bar chart
+   */
+  private createBoxplot(inputData: Item[]): void {
+    if (!this.boxplot) { return; }
+    // Remove all elements of from the previous boxplot.
+    this.boxplot.selectAll('circle').remove();
+    this.boxplot.selectAll('line').remove();
+    this.boxplot.selectAll('text').remove();
+
+    const durationSelected = inputData.map((obj: Item) => obj.durationHours);
+    const durationSorted = durationSelected.sort(d3.ascending);
+    const durationMax = durationSorted[durationSorted.length - 1];
+    const durationMin = durationSorted[0];
+
+    const q1 = d3.quantile(durationSorted, 0.25);
+    if (!q1) { return; }
+    const median = d3.quantile(durationSorted, 0.5);
+    if (!median) { return; }
+    const q3 = d3.quantile(durationSorted, 0.75);
+    if (!q3) { return; }
+    const dataLabels = [durationMax, durationMin, q1, median, q3];
+
+    const boxWidth = 30;
+    // If more than 100 Items are selected, set the circle radius to be 1.4; otherwise, set
+    // the radius to 2.5.
+    const pointRadius = (inputData.length > BarChartComponent.DEFAULT_MAX_LABELS) ? 1.4 : 2.5;
+    const jitterWidth = boxWidth - 10;
+    const center = this.width - 60;
+
+    // Add individual points with jitter.
+    this.boxplot
+      .attr('class', 'boxplot-points')
+      .selectAll('points')
+      .data(inputData)
+      .enter()
+      .append('circle')
+        .attr('cx', (d: Item) => (center - jitterWidth / 2 + Math.random() * jitterWidth))
+        .attr('cy', (d: Item) => this.yScaleFocus(d.durationHours))
+        .attr('r', pointRadius)
+        .style('fill', (d: Item) => BarChartComponent.STATE_TO_COLOR[d.state])
+        .style('fill-opacity', 0.45); // Show overlay among dataPoints.
+
```

Comment:
Make the indentation consistent for the entire file.

Reply:
Actually I did that because it seems more clear to me. The `attr` amd `style` properties belongs to `<circle>` element, so I did another indentation to indicate that they are not properties of `newBars`. Let me know which way makes more sense to me! :)